mirgecom/initializers.py

```diff
@@ -262,6 +263,124 @@ def __call__(self, x_vec, *, t=0, eos=IdealSingleGas()):
                               momentum=mom)
 
 
+class DoubleMachReflection:
+    r"""Implement the double shock reflection problem.
+
+        - Woodward and Collela
+
+    The initial condition is defined
+
+    .. math::
+
+        {\rho}(x < x_s(y,t)) &= \gamma \rho_j\\
+        {\rho}(x > x_s(y,t)) &= \gamma\\
+        {\rho}{V_x}(x < x_s(y,t)) &= u_p \cos(\pi/6)\\
+        {\rho}{V_x}(x > x_s(y,t)) &= 0\\
+        {\rho}{V_y}(x > x_s(y,t)) &= u_p \sin(\pi/6)\\
+        {\rho}{V_y}(x > x_s(y,t)) &= 0\\
+        {\rho}E(x < x_s(y,t)) &= (\gamma-1)p_j\\
+        {\rho}E(x > x_s(y,t)) &= (\gamma-1)
+
+    where the shock position is given,
+
+    .. math::
+
+        x_s = x_0 + y/\sqrt{3} + 2 u_s t/\sqrt{3}
+
+    and the normal shock jump relations are
+
+    .. math::
+
+        \rho_j &= \frac{(\gamma + 1) u_s^2}{(\gamma-1) u_s^2 + 2} \\
+        p_j &= \frac{2 \gamma u_s^2 - (\gamma - 1)}{\gamma+1} \\
+        u_p &= 2 \frac{u_s^2-1}{(\gamma+1) u_s}
+
+    The initial shock location is given by $x_0$ and $u_s$ is the shock speed.
+    This function serves as an initial condition as well as boundary conditions
+    for $t>0$
+
+    .. automethod:: __init__
+    .. automethod:: __call__
+    """
+
+    def __init__(
+            self, shock_location=1.0/6.0, shock_speed=4.0
+    ):
+        """Initialize initial condition options.
+
+        Parameters
+        ----------
+        shock_location: float
+           initial location of shock
+        shock_speed: float
+           shock speed, Mach number
+        """
+        self._shock_location = shock_location
+        self._shock_speed = shock_speed
+
+    def __call__(self, x_vec, *, t=0, eos=IdealSingleGas()):
+        r"""
+        Create the initial condition for the double Mach reflection case at
+        locations *x_vec*. Also provides boundary conditions for solution at
+        times $t>0$.
```

Comment:
Echoing my last comment regarding *t>0*, and BCs callout here should be removed.  Also, and this is mostly a nitpick, but....

This routine can be used as more than an initial condition. I disagree with documenting it as an "initial condition", and instead prefer "solution initializer".  The solution initializer is used to create the double mach reflection exact solution, the end.  The user can use that functionality how they see fit, e.g. initial condition, exact soln at boundary, etc.

Reply:
So it only generates the exact solution to the problem at t=0, at t>0 it is used as a prescribed boundary condition in doublemach-mpi.py (130-132). This provides the exact far field conditions, but not does not provide the exact solution for the entirety of the domain.

Reply:
It generates the solution at locations indicated by *x_vec* and at time indicated by *t*.  It itself does not care if you use it as a BC or as an initial solution.

Reply:
Resolved: https://github.com/illinois-ceesd/mirgecom/pull/213/commits/f3e6a6110968876e40be153abdb8fb5787b32ecd